tests/unittests/tests-uri_parser/tests-uri_parser.c

```diff
@@ -24,18 +24,19 @@
     { .uri = u, .scheme = s, .userinfo = us, .host = h, .port = po,     \
       .path = pa, .query = q, .expected = e}
 
-#define VEC_CHECK(comp)                                                 \
+#define VEC_CHECK(comp, i)                                              \
     do {                                                                \
         if (ures.comp == NULL) {                                        \
             TEST_ASSERT(validate_uris[i].comp[0] == '\0');              \
         }                                                               \
         else {                                                          \
             TEST_ASSERT_EQUAL_INT(strlen(validate_uris[i].comp),        \
                                   ures.comp##_len);                     \
-            TEST_ASSERT_EQUAL_INT(0,                                    \
-                                  memcmp(ures.comp,                     \
-                                         validate_uris[i].comp,         \
-                                         strlen(validate_uris[i].comp))); \
+            TEST_ASSERT_MESSAGE(0 ==                                    \
+                                  strncmp(ures.comp,                    \
+                                          validate_uris[i].comp,        \
+                                          strlen(validate_uris[i].comp)), \
+                                "Unexpected " # comp " member");        \
```

Comment:
good change. Could you also print `i` so we know (more or less..) which vector failed?

Reply:
or maybe even the uri-to-parse?

Reply:
Sadly, not really. The symbol `i` will just be resolved to that by the preprocessor and adding a `printf` to the unittests is most often a bad idea, as it destroys the XML generated when compiled with OUTPUT=XML :-/.

Reply:
(there are enough offenders to that as it is in the unittests)

Reply:
One could change the tests that the test vectors become test cases themselves. But that is change for another time I think ;-).

Reply:
Found a way to do it :-)